src/lib/strings.h

```diff
@@ -19,12 +19,12 @@ extern "C" {
 /**
  * Helper to use strnlen() instead of strlen.
  */
-#define str_len(_str) strnlen(_str, INT64_MAX)
+#define str_len(_str) strlen(_str)
```

Comment:
Using strlen is not allowed by the swupd coding rules, `strlen` is considered unsafe so strnlen has to be used instead. The same applies to strcmp. You can see the functions that are not allowed to be used in swupd here:
https://github.com/clearlinux/swupd-client/blob/master/test/code_analysis/warning_functions.bats#L11